src/sagas/backend.ts

```diff
@@ -199,6 +199,43 @@ function* backendSaga(): SagaIterator {
     }
   });
 
+  /**
+   * Unsubmits the submission and updates the grading overviews of the new status.
+   */
+  yield takeEvery(actionTypes.UNSUBMIT_SUBMISSION, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+    const { submissionId, overviews } = (action as actionTypes.IAction).payload;
+    const resp: Response = yield postUnsubmit(submissionId, tokens);
+    const newOverviews = (overviews as GradingOverview[]).map(overview => {
+      if (overview.submissionId === submissionId) {
+        return { ...overview, submissionStatus: 'attempted' };
+      } else {
+        return overview;
+      }
+    });
+
+    if (resp && resp.ok) {
+      yield call(showSuccessMessage, 'Unsubmit successful', 1000);
+      yield put(actions.updateGradingOverviews(newOverviews));
+    } else if (resp !== null) {
+      let errorMessage: string;
```

Comment:
This block is duplicated many times in the code. Let's maintain DRY principle: Please abstract it out into a helper function.

Reply:
should we handle it now or log an issue. I can see that the entire file benefits from a `handleError` function, but I don't this PR will be suitable to change that

Reply:
We should handle it now? I don't see it as a big change to warrant a new PR. (its just refactoring?)

Your PR is actually adding to the issue and making it worse! So it is something that needs to be fixed. May I know why you would rather log it as an issue?

Reply:
Ok, I abstracted it out